### PR TITLE
Fix 2D rotational periodicity

### LINKED ISSUE #1562: 2D-ROTATIONAL PERIODIC BOUNDARY CONDITION BUG
**Describe the bug**
During the last weeks of 2021, two different periodic flow scenarios have been investigated ( #1523 ).
While setting the periodic boundary conditions, the 2D-divergent wedge case solution seemed to be negatively affected even after the precious suggestions and corrections received in the quoted discussion. 

Consequently, different test cases have been simulated again to avoid any possible mistake.
Scenarios investigated are better explained here below:
- **Rectangular Duct** 

_2D case_ 

_3D case_ 

- **Divergent Wedge**

_2D case_ 

_3D case_ 

Meshes have been realized with ICEM CFD, then correctly converted into .cgns (adf) format.

The results obtained are the following:

- 2D Rectangular Duct (Translational Periodicity) --> **Converged**
- 3D Rectangular Duct (Translational Periodicity) --> **Converged**
- 2D Divergent Wedge (Rotational Periodicity) --> _**Diverged**_
- 3D Divergent Wedge (Rotational Periodicity) --> **Converged**

It is suggested that this solution behavior might be traced to a bug.
It has also been observed that the converged simulations are providing consistent results only for the following convective scheme and time discretization settings: 

CONV_NUM_METHOD_FLOW = ROE
MUSCL_FLOW = NO
TIME_DISCRE_FLOW = EULER_IMPLICIT

**To Reproduce the simulations**

_2D Rectangular Duct_
[channel2D.txt](https://github.com/su2code/SU2/files/8238476/channel2D.txt)

_3D Rectangular Duct_
[channel3D.txt](https://github.com/su2code/SU2/files/8238474/channel3D.txt)

_2D Divergent Wedge_
[wedge2D.txt](https://github.com/su2code/SU2/files/8238478/wedge2D.txt)

_3D Divergent Wedge_
[wedge3D.txt](https://github.com/su2code/SU2/files/8238481/wedge3D.txt)

 .cgns files cannot be attached to this report (format not supported from GitHub) but can be provided separately.
Thank you in advance for your support, always precious and more than welcome.

**Desktop:**
OS: Microsoft Windows 10 64bit
C++ compiler and version: g++ (GCC) 11.1.0
MPI implementation and version: OpenMPI 10.1.12498.18
SU2 Version: v7.2.0 Blackbird

### PR DESCRIPTION
**Pull request by pcarruscag**
Fixes #1562

### CODE REVIEW

File: TestCases/navierstokes/periodic2D/config.cfg [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+%
+CONV_RESIDUAL_MINVAL= -11
+CONV_STARTITER= 0
+INNER_ITER= 5000
+
+% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
+%
+MESH_FORMAT= SU2
+MESH_FILENAME= sector.su2
+VOLUME_OUTPUT= PRIMITIVE, RESIDUAL
```
Comment (pcarruscag) [inline TestCases/navierstokes/periodic2D/config.cfg:73]:
```suggestion
```

File: TestCases/hybrid_regression.py [hunk 2]
```diff
@@ -134,6 +134,14 @@ def main():
     poiseuille_profile.test_vals = [-12.494752, -7.712204, -0.000000, 2.085796]
     test_list.append(poiseuille_profile)
 
+    # 2D Periodic
+    periodic2d           = TestCase('periodic2d')
+    periodic2d.cfg_dir   = "navierstokes/periodic2D"
+    periodic2d.cfg_file  = "config.cfg"
+    periodic2d.test_iter = 1400
+    periodic2d.test_vals = [-10.818512, -8.363387, -8.287483, -5.334815, -1.087922, -2945.2]
```
Comment (pcarruscag) [inline TestCases/hybrid_regression.py:142]:
```suggestion
    periodic2d.test_vals = [-10.818511, -8.363385, -8.287482, -5.334813, -1.087926, -2945.2]
```

File: TestCases/navierstokes/periodic2D/sector.geo [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+//+
+Transfinite Curve {1, 2} = N Using Progression 1;
+//+
+Transfinite Curve {3, 4} = N Using Progression 1;
+//+
+Transfinite Surface {1};
+//+
+Recombine Surface {1};
+//+
+Mesh 2;
```
Comment (TobiKattmann) [inline TestCases/navierstokes/periodic2D/sector.geo:46]:
```suggestion
// Gmsh .geo script
ri = 0.25;
ro = 0.5;
N = 40;

Point(1) = {0.0, 0.0, 0.0, 1.0};
Point(2) = {ri, 0.0, 0.0, 1.0};
Point(3) = {ro, 0.0, 0.0, 1.0};
Point(4) = {ro/Sqrt(2), ro/Sqrt(2), 0.0, 1.0};
Point(5) = {ri/Sqrt(2), ri/Sqrt(2), 0.0, 1.0};

Circle(1) = {2, 1, 5};
Circle(2) = {3, 1, 4};
Line(3) = {2, 3};
Line(4) = {5, 4};

Curve Loop(1) = {4, -2, -3, 1};
Surface(1) = {1};

Physical Curve("inlet", 5) = {1};
Physical Curve("outlet", 6) = {2};
Physical Curve("per1", 7) = {3};
Physical Curve("per2", 8) = {4};
Physical Surface("fluid", 9) = {1};

Transfinite Curve {1, 2} = N Using Progression 1;
Transfinite Curve {3, 4} = N Using Progression 1;
Transfinite Surface {1};
Recombine Surface {1};

Mesh 2;
```

Comment (TobiKattmann) [reply TestCases/navierstokes/periodic2D/sector.geo:46]: Just a little chore ;)

File: SU2_CFD/src/solvers/CSolver.cpp [hunk 4]
```diff
@@ -738,6 +738,22 @@ void CSolver::InitiatePeriodicComms(CGeometry *geometry,
               Rotate(zeros, jacBlock[iVar], rotBlock[iVar]);
             }
 
+            /*--- Rotate the vector components of the solution. ---*/
+
+            if (rotate_periodic) {
+              for (iDim = 0; iDim < nDim; iDim++) {
+                su2double d_diDim[3] = {0.0};
+                for (iVar = 1; iVar < 1+nDim; ++iVar) {
+                  d_diDim[iVar-1] = rotBlock(iVar, iDim);
```
Comment (TobiKattmann) [inline SU2_CFD/src/solvers/CSolver.cpp:747]: fun with indices :)

File: TestCases/hybrid_regression.py [hunk 5]
```diff
@@ -134,6 +134,14 @@ def main():
     poiseuille_profile.test_vals = [-12.494752, -7.712204, -0.000000, 2.085796]
     test_list.append(poiseuille_profile)
 
+    # 2D Periodic
```
Comment (TobiKattmann) [inline TestCases/hybrid_regression.py:137]:
```suggestion
    # 2D rotational Periodic
```
a small additional hint

**Review by TobiKattmann — APPROVED** [review]
Thanks for this lightning quick bugfix and the additional testcase with meshing script :bouquet: 
I guess I still have to get used to having small meshes in the code repo from now on :)
